aws/retry/retryable_error_test.go

```diff
@@ -0,0 +1,337 @@
+package retry_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/awserr"
+	"github.com/aws/aws-sdk-go-v2/aws/retry"
+)
+
+// TODO add retryable tests
+
+func TestRetryConnectionErrors(t *testing.T) {
+	cases := map[string]struct {
+		Err       error
+		Retryable aws.Ternary
+	}{
+		"nested connection reset": {
+			Retryable: aws.TrueTernary,
+			Err: fmt.Errorf("serialization error, %w",
+				fmt.Errorf("connection reset")),
+		},
+		"top level connection reset": {
+			Retryable: aws.TrueTernary,
+			Err:       fmt.Errorf("connection reset"),
+		},
+		"awserr connection reset": {
+			Retryable: aws.TrueTernary,
+			Err: awserr.New(aws.ErrCodeSerialization, "some error",
+				fmt.Errorf("connection reset")),
+		},
+		"nil error connection reset": {
+			Retryable: aws.UnknownTernary,
+		},
+		"some other error": {
+			Retryable: aws.UnknownTernary,
+			Err: awserr.New(aws.ErrCodeSerialization, "some error",
+				fmt.Errorf("something bad")),
+		},
+		"request error": {
+			Retryable: aws.UnknownTernary,
+			Err: awserr.New(aws.ErrCodeSerialization, "some error",
+				fmt.Errorf("something bad")),
+		},
+	}
+
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
+			var r retry.RetryableConnectionError
+
+			retryable := r.IsErrorRetryable(c.Err)
+			if e, a := c.Retryable, retryable; e != a {
+				t.Errorf("expect %v retryable, got %v", e, a)
+			}
+		})
+	}
+}
+
+func TestRetryHTTPStatusCodes(t *testing.T) {
+}
+
+//func TestIsSerializationErrorRetryable(t *testing.T) {
+//	testCases := []struct {
+//		err      error
+//		expected bool
+//	}{
```

Comment:
Did you mean to remove these commented test cases, or were you going to follow-up with additional ones?

Reply:
these are cases copied from the old retry behavior. Commented our at while i port them over to this new logic.